matsim/src/main/java/org/matsim/core/mobsim/hermes/HermesBuilder.java

```diff
@@ -23,36 +23,55 @@
 import org.matsim.core.controler.AbstractModule;
 import org.matsim.core.controler.AllowsConfiguration;
 import org.matsim.core.mobsim.qsim.AbstractQSimModule;
+import org.matsim.core.mobsim.qsim.components.QSimComponentsConfig;
 import org.matsim.core.mobsim.qsim.components.QSimComponentsConfigurator;
+import org.matsim.core.router.util.TravelTime;
 
-class HermesBuilder implements AllowsConfiguration {
+import java.util.*;
+
+public class HermesBuilder implements AllowsConfiguration {
+	private final Collection<AbstractQSimModule> qsimModules = new LinkedList<>();
+	private final QSimComponentsConfig components = new QSimComponentsConfig();
+
+	private final List<AbstractModule> overridingControllerModules = new LinkedList<>();
+	private final List<AbstractQSimModule> overridingQSimModules = new LinkedList<>();
+	private final Map<String, TravelTime> travelTimes = new HashMap<String, TravelTime>();
 
 	@Override
-	public AllowsConfiguration addOverridingModule(AbstractModule abstractModule) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder addOverridingModule(AbstractModule abstractModule) {
+		overridingControllerModules.add(abstractModule);
+		return this;
 	}
 
 	@Override
-	public AllowsConfiguration addOverridingQSimModule(AbstractQSimModule qsimModule) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder addOverridingQSimModule(AbstractQSimModule qsimModule) {
+		this.overridingQSimModules.add(qsimModule);
+		return this;
 	}
 
 	@Override
-	public AllowsConfiguration addQSimModule(AbstractQSimModule qsimModule) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder addQSimModule(AbstractQSimModule qsimModule) {
+		this.qsimModules.add(qsimModule);
+		return this;
 	}
 
 	@Override
-	public AllowsConfiguration configureQSimComponents(QSimComponentsConfigurator configurator) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder configureQSimComponents(QSimComponentsConfigurator configurator) {
+		configurator.configure(components);
+		return this;
 	}
-	
+
+	public HermesBuilder addTravelTime(String mode, TravelTime travelTime) {
+		this.travelTimes.put(mode, travelTime);
+		return this;
+	}
+
 	public Hermes build(Scenario scenario, EventsManager eventsmanager) {
```

Comment:
I think you could get some unit testing around this builder class too, for example to check that the Hermes instance created has the correct travel times. Again we're back at the question of how to get at those travel times. I'm not really a fan of making public things that should be `private`, so maybe we need an accessor method or something? Seems like it would unlock a lot of unit testing.